### ISSUE TITLE
thread '<unnamed>' panicked at 'On macOS, `EventLoop` must be created on the main thread!'

### ISSUE THREAD

**Issue by declanqian** [emacs-ng/emacs-ng#366]
```
thread '<unnamed>' panicked at 'On macOS, `EventLoop` must be created on the main thread!', /Users/declan/.cargo/registry/src/github.com-1ecc6299db9ec823/winit-0.24.0/src/platform_impl/macos/event_loop.rs:60:17
stack backtrace:
   0: std::panicking::begin_panic
   1: winit::event_loop::EventLoop<T>::with_user_event
   2: webrender::output::Output::create_webrender_window::{{closure}}
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
```

**Comment by declanqian** [comment]
Hey @harryfei, does it make sense to move event_loop to main thread?
I am looking at winit's [multi-threaded](https://github.com/rust-windowing/winit/blob/master/examples/multithreaded.rs) [multi-window](https://github.com/rust-windowing/winit/blob/master/examples/multiwindow.rs) example, and [servo app.rs](https://github.com/servo/servo/blob/master/ports/winit/app.rs)
all initialized an event_loop instance at main thread?

For the record, I don't have experience with multi-thread programming, C programming nor computer graphics.
I just started to get familiar with rust. And Emacs NG source is kind of complicated for me to proceed.
There are `wrterm.h`(cbindgen?) and `bindings.rs`(bindgen?) which are hurting my brain.

Also I think https://github.com/emacs-ng/emacs-ng/issues/344 may also be solved by this.

I need help from our webrender team to validate this, maybe write some rust prototype code for
initializing a global event_loop from `wr_term_init` which can be accessible when `create_webrender_window`?

**Comment by harryfei** [comment]
Yeah, you are right. 

1. I used the event_loop in a incorrect manner, so it cause the issue #344.
2. Because of macOS platform limit, we **must** run `winit`'s event loop in main thread. But emacs run it's own in main thread which is not easy to change. We need merge the two event loop into one.

Actually, I'm working on moving the event loop to main thread. But I faced with some problems because of the above 2.
